pipeline-model-definition/src/test/resources/postStage/localAll.groovy

```diff
@@ -35,6 +35,11 @@ pipeline {
                     if (res != null) {
                         echo "Setting build result ${res}"
                         currentBuild.result = res
+                        // JENKINS-52114 - can't tell the difference between setting currentBuild.result in this stage
+                        // and an error in a parallel stage, so let's error.
+                        if (res == "FAILURE") {
+                            error "Failing explicitly"
```

Comment:
Is this necessary to fix a failing test? If so, would this PR break existing Pipelines that are doing something similar by changing which post conditions are evaluated, or would it do something else?

Reply:
I don't really have an informed opinion either way, it just seems a little scary to intentionally change behavior that we were testing against. It would be nice if we had telemetry to give us an idea of how often users rely on the previous behavior.

Reply:
Oh, yeah, I'm wary of it. But I don't think it's hugely common - you can only set the build to `FAILURE` directly from within a `script` block (or I guess via some step that sets the build status to `FAILURE` without the step itself failing - the only examples I can think of for that would maybe be static analysis/code coverage publishers under certain configurations), and then you have to have `failure` in `post` for that stage.

This does merit further discussion/thought, for sure.